core-bundle/src/Resources/contao/library/Contao/Automator.php

```diff
@@ -144,18 +145,31 @@ public function purgeScriptCache()
 
 	/**
 	 * Purge the page cache
-	 *
-	 * @todo Replace this with a more sophisticated invalidation routine
 	 */
 	public function purgePageCache()
 	{
-		$strCacheDir = \StringUtil::stripRootDir(\System::getContainer()->getParameter('kernel.cache_dir'));
+		$container = \System::getContainer();
 
-		$objFolder = new \Folder($strCacheDir . '/http_cache');
-		$objFolder->purge();
+		if (!$container->has('fos_http_cache.cache_manager'))
```

Comment:
Shouldn't this case be handled before showing the option in the back end at all?

Reply:
I cannot. The maintenance module does not allow to have such pre-checks and I also don't think we should implement it because out-of-the-box it works. I only doesn't work if the configuration was adjusted and then I think it's better to log something than to just hide the option all together.